### PR TITLE
Button block to the landing page added

### LINKED ISSUE #2: Landing pages should have a block for a button
Currently there is no block in the Landing Page which allows a user to add a button.

As this is a common task that we do in landing pages, we should allow the addition of a button.

### PR DESCRIPTION
**Pull request by disha1611**
This PR adds a block for the landing page which allows you to put a button on the page.

Fixes #2

### CODE REVIEW

**Comment by RCheesley** [comment]
Thanks for the PR @disha-pishavadia24 and welcome to the community! 🎉

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@disha-pishavadia24 I think you may have pushed the pref centre work to this PR - we may need to revert that commit.

**Comment by RCheesley** [comment]
I have force-pushed the revert as it was blocking us from testing! :)

**Comment by RCheesley** [comment]
@disha-pishavadia24 with your PR checked out and then running the npm run build I am not seeing any button in the blocks that are available.

Are there some steps I am missing?  Is there some way I can check if the PR has been successfully applied (maybe inspecting the JS somehow?)

I am following these steps: https://github.com/mautic/grapesjs-preset-mautic#how-to-test-a-preset-pull-request with Adrian's PR checked out in mautic/mautic, your PR checked out in mautic/grapesjs-preset-mautic.

screenshot-mautic4 ddev site-2021 07 22-15_34_07

**Review by RCheesley — DISMISSED** [review]
Ignore my earlier comment - I was running the build command in the preset rather than the plugin which was removing changes 😊 

Looks great, works great!
  
screenshot-mautic4 ddev site-2021 07 22-15_44_09

screenshot-mautic4 ddev site-2021 07 22-15_45_33

Thanks for your work on this! 🎉

**Comment by RCheesley** [comment]
@All-contributors please add @disha-pishavadia24 for code

File: dist/blocks.js [hunk 1]
```diff
@@ -11,6 +13,35 @@ export default ((editor, opts = {}) => {
   } // Add icon to mj-hero
 
 
+  //add button block for landing page
```
Comment (adiux) [inline dist/blocks.js]: It looks to me like you added your changes to the dist/ folder? Please copy them to the file in the src/ folder. If you run `npm run build` the JS files will be built and copied to the dist/ folder.

Comment (disha1611) [reply dist/blocks.js]: Let me fix this soon.

Comment (disha1611) [reply dist/blocks.js]: I have removed the one from dist folder

File: src/blocks.js [hunk 2]
```diff
@@ -13,6 +14,35 @@ export default (editor, opts = {}) => {
     dcb.addDynamciContentBlock();
   }
 
+  //add button block for landing page 
+  const mode = ContentService.getMode(editor);   
+  if (mode === ContentService.modePageHtml) {   
+    bm.add('button', {
+      label: 'Button',
```
Comment (adiux) [inline src/blocks.js]: could be improved by making it translatable

**Comment by adiux** [comment]
@disha-pishavadia24 I think it is cleaner if we have one single file per block. Since it is simpler than the dynamicContent block, I also think it is to much to have a folder. But I think one file per block is more consistent and better to maintain in the long run.

File: src/blocks.js (hunk 2 above)
Comment (disha1611) [reply src/blocks.js]: Okay @adiux I will make the change as now I know where can I define the translatable labels. Thanks!

**Comment by disha1611** [comment]
> @disha-pishavadia24 I think it is cleaner if we have one single file per block. Since it is simpler than the dynamicContent block, I also think it is to much to have a folder. But I think one file per block is more consistent and better to maintain in the long run.

Okay Sure, thanks for the suggestion will add the new file for block and commit again.

File: src/blocks.js (hunk 2 above)
Comment (disha1611) [reply src/blocks.js]: @adiux I have made necessary changes in the files, need to add labels to the translate En and Fr folder grapesjsbuilder.buttonBlockLabel="Button" for En and grapesjsbuilder.buttonBlockLabel="Bouton" for Fr.

**Comment by disha1611** [comment]
@adiux I have added label for button in all language in Transaction folder and create a pull on mautic repo https://github.com/mautic/mautic/pull/10303  I think this will help you to review this PR. Thanks!

File: src/buttonBlock.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+export default class ButtonBlock {
+  editor;
```
Comment (adiux) [inline src/buttonBlock.js]: the two class properties: editor and opts are not needed and can be removed for a cleaner code.

Comment (disha1611) [reply src/buttonBlock.js]: @adiux you can check now.

File: src/buttonBlock.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+export default class ButtonBlock { 
+
+  blockManager;
+
+  constructor(editor, opts = {}) {
```
Comment (adiux) [inline src/buttonBlock.js]: I think the `opts` parameter is not needed and can be removed

Comment (disha1611) [reply src/buttonBlock.js]: Removed @adiux

**Comment by RCheesley** [comment]
@adiux are we good to go with this PR now, or are there any outstanding feedbacks that need further work?

**Comment by RCheesley** [comment]
@disha-pishavadia24 can you take a look at the failing tests? Looks like a code style issue.

**Review by RCheesley — APPROVED** [review]

**Pull request merged**